// shelljs/shell.js
//
// ShellJS
// Unix shell commands on top of Node's API
//
// Copyright (c) 2012 Artur Adib
// http://github.com/arturadib/shelljs
//

var common = require('./lib/common');


//@
//@ All commands run synchronously, unless otherwise stated.
//@

//@include ./lib/cd
var _cd = require('./lib/cd');
exports.cd = common.wrap('cd', _cd);

//@include ./lib/pwd
var _pwd = require('./lib/pwd');
exports.pwd = common.wrap('pwd', _pwd);

//@include ./lib/ls
var _ls = require('./lib/ls');
exports.ls = common.wrap('ls', _ls);

//@include ./lib/find
var _find = require('./lib/find');
exports.find = common.wrap('find', _find);

//@include ./lib/cp
var _cp = require('./lib/cp');
exports.cp = common.wrap('cp', _cp);

//@include ./lib/rm
var _rm = require('./lib/rm');
exports.rm = common.wrap('rm', _rm);

//@include ./lib/mv
var _mv = require('./lib/mv');
exports.mv = common.wrap('mv', _mv);

//@include ./lib/mkdir
var _mkdir = require('./lib/mkdir');
exports.mkdir = common.wrap('mkdir', _mkdir);

//@include ./lib/test
var _test = require('./lib/test');
exports.test = common.wrap('test', _test);

//@include ./lib/cat
var _cat = require('./lib/cat');
exports.cat = common.wrap('cat', _cat);

//@include ./lib/to
var _to = require('./lib/to');
String.prototype.to = common.wrap('to', _to);

//@include ./lib/toEnd
var _toEnd = require('./lib/toEnd');
String.prototype.toEnd = common.wrap('toEnd', _toEnd);

//@include ./lib/sed
var _sed = require('./lib/sed');
exports.sed = common.wrap('sed', _sed);

//@include ./lib/grep
var _grep = require('./lib/grep');
exports.grep = common.wrap('grep', _grep);

//@include ./lib/which
var _which = require('./lib/which');
exports.which = common.wrap('which', _which);

//@include ./lib/echo
var _echo = require('./lib/echo');
exports.echo = _echo; // don't common.wrap() as it could parse '-options'

//@include ./lib/dirs
var _dirs = require('./lib/dirs').dirs;
exports.dirs = common.wrap("dirs", _dirs);
var _pushd = require('./lib/dirs').pushd;
exports.pushd = common.wrap('pushd', _pushd);
var _popd = require('./lib/dirs').popd;
exports.popd = common.wrap("popd", _popd);

//@include ./lib/ln
var _ln = require('./lib/ln');
exports.ln = common.wrap('ln', _ln);

//@
//@ ### exit(code)
//@ Exits the current process with the given exit code.
exports.exit = process.exit;

//@
//@ ### env['VAR_NAME']
//@ Object containing environment variables (both getter and setter). Shortcut to process.env.
exports.env = process.env;

//@include ./lib/exec
var _exec = require('./lib/exec');
exports.exec = common.wrap('exec', _exec, {notUnix:true});

//@include ./lib/chmod
var _chmod = require('./lib/chmod');
exports.chmod = common.wrap('chmod', _chmod);



//@
//@ ## Non-Unix commands
//@

//@include ./lib/tempdir
var _tempDir = require('./lib/tempdir');
exports.tempdir = common.wrap('tempdir', _tempDir);


//@include ./lib/error
var _error = require('./lib/error');
exports.error = _error;



//@
//@ ## Configuration
//@

exports.config = common.config;

//@
//@ ### config.silent
//@ Example:
//@
//@ ```javascript
//@ var sh = require('shelljs');
//@ var silentState = sh.config.silent; // save old silent state
//@ sh.config.silent = true;
//@ /* ... */
//@ sh.config.silent = silentState; // restore old silent state
//@ ```
//@
//@ Suppresses all command output if `true`, except for `echo()` calls.
//@ Default is `false`.

//@
//@ ### config.fatal
//@ Example:
//@
//@ ```javascript
//@ require('shelljs/global');
//@ config.fatal = true;
//@ cp('this_file_does_not_exist', '/dev/null'); // dies here
//@ /* more commands... */
//@ ```
//@
//@ If `true` the script will die on errors. Default is `false`.
